### PR TITLE
lib/storage: new storage search benchmarks

### PR DESCRIPTION
**Pull request by rtm0**
### Describe Your Changes

New benchmarks for storage search (data and index):
- Use the same dataset that accounts for prev and curr indexDBs and deleted series
- The code is more structured
- Account for various numbers of series in response including higher numbers (>10k) as this appears to be a quite common use case.

These bechmarks were used for investigating #9602 performance issue  and helped discover that prefetching metric names needed to be restored #9619.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres to [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/victoriametrics/contributing/#pull-request-checklist).
- [x] My change adheres to [VictoriaMetrics development goals](https://docs.victoriametrics.com/victoriametrics/goals/).

### CODE REVIEW

File: lib/storage/storage_timing_test.go [hunk 1]
```diff
@@ -612,3 +288,416 @@ func benchmarkDirSize(path string) int64 {
...
+// benchmarkSearch_variableSeries measures the execution time of some search
+// operation on a fixed time trange and variable number of series. The number of
+// deleted series is 0.
+func benchmarkSearch_variableSeries(b *testing.B, graphite bool, op func(b *testing.B, s *Storage, tr TimeRange, mrs []MetricRow)) {
+	const numDeletedSeries = 0
+	tr := TimeRange{
+		MinTimestamp: time.Date(2025, 1, 1, 0, 0, 0, 0, time.UTC).UnixMilli(),
+		MaxTimestamp: time.Date(2025, 1, 1, 23, 59, 59, 999_999_999, time.UTC).UnixMilli(),
+	}
+	for _, numSeries := range []int{
```
Comment (f41gh7) [inline lib/storage/storage_timing_test.go]: This number looks to broad. I'd suggest to reduce step. Added benchmarks took a lot of time to complete. 

 I'd suggest to provide 3-4 cases.

File: lib/storage/storage_timing_test.go [hunk 2]
```diff
@@ -612,3 +288,416 @@ func benchmarkDirSize(path string) int64 {
...
+
+// benchmarkSearch_variableDeletedSeries measures the execution time of some
+// storage operation on a fixed time, variable number of series and variable
+// number of deleted series.
+func benchmarkSearch_variableDeletedSeries(b *testing.B, graphite bool, op func(b *testing.B, s *Storage, tr TimeRange, mrs []MetricRow)) {
+	tr := TimeRange{
+		MinTimestamp: time.Date(2025, 1, 1, 0, 0, 0, 0, time.UTC).UnixMilli(),
+		MaxTimestamp: time.Date(2025, 1, 1, 23, 59, 59, 999_999_999, time.UTC).UnixMilli(),
+	}
+	for _, numSeries := range []int{100, 1000, 10_000, 100_000, 1_000_000} {
```
Comment (f41gh7) [inline lib/storage/storage_timing_test.go]: This combination of test cases also could be reduced to only cover edge cases.

File: lib/storage/storage_timing_test.go (hunk 1 above)
Comment (rtm0) [reply lib/storage/storage_timing_test.go]: Reduced to a 5 numbers that represent orders of magnitude.

File: lib/storage/storage_timing_test.go (hunk 2 above)
Comment (rtm0) [reply lib/storage/storage_timing_test.go]: Replaced the outer loop with just one value of 100k that should more or less reflect the real number of series in request result.

**Comment by rtm0** [comment]
@f41gh7 Reduced the number of cases, PTAL.

**Review by f41gh7 — APPROVED** [review]
LGTM
